### PR TITLE
feat: Enable setting the Prometheus HTTP metrics endpoint

### LINKED ISSUE #5: Add support for `admin.metrics_endpoint`
Stunner has a new admin.metrics_endpoint parameter: https://github.com/l7mp/stunner/blob/main/cmd/stunnerd/stunnerd.conf#L5

This configuration parameter is unknown to the operator. As consequence, the operator clears this parameter at an update. This blocks the stunner monitoring feature.

### PR DESCRIPTION
**Pull request by levaitamas**
This PR introduces the MetricsEndpoint configuration parameter. This parameter sets the Prometheus metrics HTTP endpoint of STUNner.

~~**Note:** This is still a draft PR: it depends on a stunner version update (and a go.mod update), and perhaps a code review.~~

Fixes #5.

### CODE REVIEW

**Comment by levaitamas** [comment]
Note: linter fails due to temporary dev hack in [go.mod](https://github.com/l7mp/stunner-gateway-operator/blob/admin-metricsendpoint/go.mod#L20).

File: go.mod [hunk 1]
```diff
@@ -17,7 +17,7 @@ require (
 	sigs.k8s.io/gateway-api v0.4.3
 )
 
-replace github.com/l7mp/stunner => ../stunner
+replace github.com/l7mp/stunner => github.com/l7mp/stunner v0.10.9-0.20220928203107-a2d718d90804
```
Comment (rg0now) [inline go.mod:20]: I'm so sorry, this was my fault, forgot to tag a release. STUNner v.0.10.9 is now tagged, could you please update the dependency accordingly?

**Review by rg0now — CHANGES_REQUESTED** [review]

File: api/v1alpha1/groupversion_info.go [hunk 2]
```diff
@@ -15,8 +15,8 @@ limitations under the License.
 */
 
 // Package v1alpha1 contains API Schema definitions for the stunner v1alpha1 API group
-//+kubebuilder:object:generate=true
-//+groupName=stunner.l7mp.io
+// +kubebuilder:object:generate=true
```
Comment (rg0now) [inline api/v1alpha1/groupversion_info.go:18]: I'm wondering what the whitespace  convention is supposed to be here...

File: config/crd/bases/stunner.l7mp.io_gatewayconfigs.yaml [hunk 3]
```diff
@@ -69,6 +69,10 @@ spec:
                   STUNner relay connections.
                 format: int32
                 type: integer
+              metricsEndpoint:
```
Comment (rg0now) [inline config/crd/bases/stunner.l7mp.io_gatewayconfigs.yaml:72]: CRD changes so unfortunately there are some further steps needed:
* run `make manifests` and `make generate`, and make sure the manifests are indeed updated (should appear here in the changed files list, or did I miss smthg?) worth issuing a `make deploy` into a working cluster and give it a try too  
* Helm chart must be updated: clone https://github.com/l7mp/stunner-helm, copy the new gatewayconfig CRD there: once we tag the release in the operator side, the helm chart gets automatically updated and people get our new CRD
* Update the docs here: https://github.com/l7mp/stunner/blob/main/doc/GATEWAY.md#gatewayconfig

Comment (rg0now) [reply config/crd/bases/stunner.l7mp.io_gatewayconfigs.yaml:72]: Oh, and don't forget to use the new git message format, see the discord

**Review by rg0now — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (levaitamas) [reply go.mod]: done in [5d90990](https://github.com/l7mp/stunner-gateway-operator/pull/6/commits/5d909909f32b8b8703710555d469c7a6daf98ad9)

**Comment by levaitamas** [comment]
Related PR:
Helm update: https://github.com/l7mp/stunner-helm/pull/3
Stunner doc update: TODO

File: config/crd/bases/stunner.l7mp.io_gatewayconfigs.yaml (hunk 3 above)
Comment (levaitamas) [reply config/crd/bases/stunner.l7mp.io_gatewayconfigs.yaml:87]: Thanks!

Please check these related PRs:
Helm update: https://github.com/l7mp/stunner-helm/pull/3
Stunner doc update: https://github.com/l7mp/stunner/pull/41

**Pull request merged**